### PR TITLE
Remove unifyEnvVarExpansion workaround for Prometheus receiver

### PR DESCRIPTION
**Pull request by pavolloffay**
**Description:** <Describe what has changed.>

**Link to tracking Issue(s):** <Issue number if applicable>

- Resolves: #3142 

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: pkg/collector/upgrade/v0_104_0.go [hunk 1]
```diff
@@ -37,3 +38,34 @@ func upgrade0_104_0(u VersionUpgrade, otelcol *v1beta1.OpenTelemetryCollector) (
 	u.Recorder.Event(otelcol, "Warning", "Upgrade", warnStr)
 	return otelcol, nil
 }
+
+// TAUnifyEnvVarExpansion disables confmap.unifyEnvVarExpansion featuregate on
+// collector instances if a prometheus receiver is configured.
+// NOTE: We need this for now until 0.105.0 is out with this fix:
+// https://github.com/open-telemetry/opentelemetry-collector/commit/637b1f42fcb7cbb7ef8a50dcf41d0a089623a8b7
+func TAUnifyEnvVarExpansion(otelcol *v1beta1.OpenTelemetryCollector) {
```
Comment (iblancasa) [inline pkg/collector/upgrade/v0_104_0.go:46]: Should we add something in the upgrade routine to 0.105.0 to remove the flag if it exists?

Comment (jaronoff97) [reply pkg/collector/upgrade/v0_104_0.go:46]: +1

**Comment by pavolloffay** [comment]
@frzifus could you please review?

**Review by yuriolisa — APPROVED** [review]

File: pkg/collector/upgrade/v0_105_0.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	}
+	if !strings.Contains(featureGates, feature) {
+		return args
+	}
+
+	features := strings.Split(featureGates, ",")
+	features = slices.DeleteFunc(features, func(s string) bool {
+		return s == feature
+	})
+	fmt.Println(features)
```
Comment (jaronoff97) [inline pkg/collector/upgrade/v0_105_0.go:59]: we should remove this

Comment (pavolloffay) [reply pkg/collector/upgrade/v0_105_0.go]: done

**Review by jaronoff97 — APPROVED** [review]

**Review by swiatekm — APPROVED** [review]

**Review by frzifus — COMMENTED** [review]
Thanks, LGTM!

**Pull request merged**